File: source/backend-vacatures-bemiddelaar/src/main/java/com/cap/pocvng/entity/ContactpersoonAfdeling.java
package com.cap.pocvng.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.ElementCollection;
import javax.validation.Valid;
import javax.validation.constraints.Size;
import java.util.List;

/**
 * ContactpersoonAfdeling
 */


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ContactpersoonAfdeling {

    @Size(max = 35)
    private String naamContactpersoonAfdeling;

    @Valid
    @ElementCollection
    private List<Telefoonummer> telefoonnummer;

    @Valid
    @ElementCollection
    private List<Emailadres> emailadres;

}

File: source/frontend/src/app/entities/werkzoekende/contactpersoonAfdeling.ts
import {Emailadres} from '../shared/emailadres';
import {Telefoonnummer} from '../shared/telefoonnummer';

export interface ContactpersoonAfdeling {
  naamContactpersoonAfdeling?: string;
  telefoonnummer?: Telefoonnummer;
  emailadres?: Emailadres;
}
